File: hireready-backend/app/schema/job.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from uuid import UUID

class JobCreate(BaseModel):
    title: str
    description: str
    required_skills: List[str] = []
    experience_level: str
    work_location: str
    employment_type: str

class JobResponse(BaseModel):
    id: UUID
    recruiter_id: UUID
    title: str
    description: str
    required_skills: List[str]
    experience_level: str
    work_location: str
    employment_type: str
    created_at: datetime
    applicant_count: int = 0
    avg_match_score: float = 0.0

    class Config:
        from_attributes = True

class ApplicationResponse(BaseModel):
    id: UUID
    resume_url: str
    status: str
    applied_at: datetime
    match_score: float
    matched_skills: List[str]
    missing_skills: List[str]
    job: JobResponse

class JobApplicantResponse(BaseModel):
    id: UUID
    job_id: UUID
    job_title: str
    seeker_id: UUID
    seeker_name: str
    seeker_email: str
    resume_url: str
    status: str
    match_score: float
    matched_skills: List[str]
    missing_skills: List[str]
    applied_at: datetime

File: hireready-frontend/lib/types/job.ts
export interface JobMetadata {
  roles: string[];
  skills: string[];
}

export interface JobCreateRequest {
  title: string;
  description: string;
  required_skills: string[];
  experience_level: string;
  work_location: string;
  employment_type: string;
}

export interface JobResponse {
  id: string;
  recruiter_id: string;
  title: string;
  description: string;
  required_skills: string[];
  experience_level: string;
  work_location: string;
  employment_type: string;
  created_at: string;
  applicant_count: number;
  avg_match_score: number;
}

export interface JobApplicantResponse {
  id: string;
  job_id: string;
  job_title: string;
  seeker_id: string;
  seeker_name: string;
  seeker_email: string;
  resume_url: string;
  status: string;
  match_score: number;
  matched_skills: string[];
  missing_skills: string[];
  applied_at: string;
}

// Deprecated or legacy frontend-only type if still needed elsewhere
export interface JobPosting {
  id: string;
  title: string;
  description: string;
  requiredSkills: string[];
  experienceLevel: string;
  workLocation: string;
  employmentType: string;
  recruiterId: string;
  createdAt: string;
  applicantCount?: number;
  avgMatch?: number;
}
